### PR TITLE
step function integration for R binding

### PR DESCRIPTION
**Pull request by jasonge27**
Add step functions support for Metaflow R binding.

### CODE REVIEW

File: R/R/flags.R [hunk 1]
```diff
@@ -38,6 +38,26 @@ parse_arguments <- function(arguments = NULL) {
...
       next
     }
+
+    if (argument == "step-functions"){
+      i <- i + 1
+      if (i <= n){
+        if (arguments[i] == "trigger"){
+          i <- i + 1
+          if (i <= n){
+            values$step_functions <- paste("trigger", arguments[i])
```
Comment (savingoyal) [inline R/R/flags.R:48]: What's the rationale for this conditional?

File: R/R/run.R [hunk 2]
```diff
@@ -90,6 +90,28 @@ run_cmd <- function(flow_file, ...) {
     batch <- ""
   }
 
+  if ("step_functions" %in% names(flags)) {
+    sfn_cmd <- paste("step-functions", flags$step_functions)
+    for (subcommand in c("authorize", "generate_new_token", 
+                         "only_json", "running", "succeeded",
```
Comment (savingoyal) [inline R/R/run.R:96]: `running`, `succeeded`, `failed`, `timed_out`, `aborted` are args for the subcommand `step-functions list-runs`. It is not immediately obvious how we are processing a command say - `step-functions list-runs running`

Comment (jasonge27) [reply R/R/run.R:96]: The user can do `Rscript flow.R step-functions list-runs --running`. Or 
```
run(step_functions = "list_runs", running=TRUE)
```

File: R/R/flags.R (hunk 1 above)
Comment (jasonge27) [reply R/R/flags.R:48]: Misread the trigger options. Will remove the conditional and submit a change.

File: metaflow/plugins/aws/step_functions/step_functions.py [hunk 3]
```diff
@@ -606,7 +607,11 @@ def _step_cli(self,
 
         script_name = os.path.basename(sys.argv[0])
         executable = self.environment.executable(node.name)
-        entrypoint = [executable, script_name]
+
+        if R.use_r():
```
Comment (savingoyal) [inline metaflow/plugins/aws/step_functions/step_functions.py:611]: @jasonge27 Can we make this change a separate PR so that we can merge and make a Pythonic release for Metaflow?

Comment (jasonge27) [reply metaflow/plugins/aws/step_functions/step_functions.py:611]: Yes here's the PR https://github.com/Netflix/metaflow/pull/317  @savingoyal

**Review by savingoyal — APPROVED** [review]
LGTM! Tested and verified!
